packages/appsync-modelgen-plugin/src/utils/process-primary-key.ts

```diff
@@ -0,0 +1,17 @@
+import { CodeGenModel } from '../visitors/appsync-visitor';
+import { getDirective } from './fieldUtils';
+
+export const processPrimaryKey = (model: CodeGenModel) => {
+  const primaryKeyField = model.fields.find(field => getDirective(field)('primaryKey'));
```

Comment:
Do you know if there is a validation check already in place to make sure we have only one `primaryKey` on a model?

Reply:
Yes, there is.

Reply:
not that I'm aware of but I can add a sanity check here

Reply:
@phani-srikar was your question about codegen or the `@primaryKey` directive? There is a [check](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts#L125) and a regression test in the `@primaryKey` implementation.

Reply:
I meant on the codegen side. Better to have some validation in place. Thanks Colin!